### PR TITLE
CDAC OTP sending support added (#51)

### LINKED ISSUE #51: [BE] Use CDAC service instead of Gupshup
## Description

We shall use [CDAC service](https://github.com/Samarth-HP/CDAC-Service) instead of Gupshup(paid service) for sending all types of SMS.

### PR DESCRIPTION
**Pull request by choxx**
## Description

Added OTP sending support via CDAC too (can be configured via environment variable `SMS_ADAPTER_TYPE`)

Closes #51

### CODE REVIEW

File: src/api/sms/cdac/cdac.service.ts [hunk 1]
```diff
@@ -1,82 +1,196 @@
...
-    for (let i = 0; i < message.length; i++) {
-      const ch = message.charCodeAt(i);
-      const j = ch;
-      const sss = '&#' + j + ';';
-      finalMessage = finalMessage + sss;
+  private __get_secret(phone): string {
+    return `${this.configService.get<string>('SMS_TOTP_SECRET')}${phone}`
+  }
+
+  private doOTPRequest(data: SMSData): Promise<any> {
```
Comment (yuvrajsab) [inline src/api/sms/cdac/cdac.service.ts:56]: why are we using "__methodname" convention here for private methods in '__get_secret' but not in 'doOTPRequest'

**Review by yuvrajsab — CHANGES_REQUESTED** [review]
also we should format our code with `yarn format` command to keep it consistent.

File: src/api/sms/cdac/cdac.service.ts [hunk 2]
```diff
@@ -1,82 +1,196 @@
...
+  SMSType,
+  TrackResponse,
+} from '../sms.interface';
 
-import { Injectable } from '@nestjs/common';
+import { HttpException, Injectable } from '@nestjs/common';
 import { SmsService } from '../sms.service';
-import fetch from 'node-fetch';
+import { ConfigService } from '@nestjs/config';
+import got, { Got } from 'got/dist/source';
```
[the remark this answers is not in the document]
Comment (choxx) [reply src/api/sms/cdac/cdac.service.ts]: The snippet was copied from legacy codes in other services. Corrected the import now.

File: src/api/sms/cdac/cdac.service.ts (hunk 1 above)
Comment (choxx) [reply src/api/sms/cdac/cdac.service.ts]: No specific reason; but good to keep using the same code formats across the code base. Corrected now.

**Review by yuvrajsab — APPROVED** [review]
lgtm

**Pull request merged**